app/controllers/assignments_controller.rb

```diff
@@ -659,19 +672,6 @@ def get_file_info(file_name, file, assignment_id, path)
     }
   end
 
-    def update_assignment!(map)
```

Comment:
@ashih2018 okay I finally understand the problem. It was an error introduced by a previous refactoring that was never caught because we didn't test yml uploads here (so it's great that you're adding to this).

Basically you should delete this method entirely (i.e., don't just move it, delete it), and do the following:

1. Move the code that deals with creating and updating the assignment directly into the body of the loop in the `Assignment.upload_assignment_list` method. Note that the `submission_rule` and `assignment_stat` attributes seem to be set both in that loop and in this method here, so you only need to keep one.

2. *Don't* worry about keeping the `flash_message` call, you can just delete that.

3. You can delete the `@testing_controller` attribute you added in the `upload_assignment_list` method, since you'll no longer need a controller instance. We never should need to use controller methods inside a model.